sklearn/gaussian_process/tests/test_gpr.py

```diff
@@ -360,3 +362,42 @@ def test_K_inv_reset(kernel):
     gpr2.predict(X2, return_std=True)
     # the value of K_inv should be independent of the first fit
     assert_array_equal(gpr._K_inv, gpr2._K_inv)
+
+
+def test_warning():
+    kernel = RBF(length_scale_bounds=[1e-5, 1e-3])
+    gpr = GaussianProcessRegressor(kernel=kernel)
+    assert_warns_message(ConvergenceWarning, "The optimal value found for "
+                                             "dimension 0 of parameter "
+                                             "length_scale is close to "
+                                             "the specified lower bound "
+                                             "1e-05. Decreasing the bound "
+                                             "and calling fit again may "
+                                             "find a better value.",
+                         gpr.fit, X, y)
+
+    kernel_sum = \
+        WhiteKernel(noise_level_bounds=[1e-5, 1e-3]) + \
+        RBF(length_scale_bounds=[1e-5, 1e-3])
+    gpr_sum = GaussianProcessRegressor(kernel=kernel_sum)
+    assert_warns_message(ConvergenceWarning, "The optimal value found for "
+                                             "dimension 0 of parameter "
+                                             "k2__length_scale is close to "
+                                             "the specified lower bound "
+                                             "1e-05. Decreasing the bound "
+                                             "and calling fit again may "
+                                             "find a better value.",
+                         gpr_sum.fit, X, y)
```

Comment:
For this example (and the one above), two warnings will actually be prompted, I don't know if there is a clean way to use `assert_warns_message` for multiple warnings ?